next/src/validation/json-logic.ts

```diff
@@ -1,28 +1,30 @@
 import type { ValidationError, ValidationErrorPath } from '../errors'
-import type { JsonLogicContext, NonBooleanJsfSchema } from '../types'
+import type { JsonLogicContext, NonBooleanJsfSchema, SchemaValue } from '../types'
+import type { ValidationOptions } from './schema'
 import jsonLogic from 'json-logic-js'
+import { validateSchema } from './schema'
 
 /**
- * (Ported from v0. TODO: check why we need it and if the name is correct)
- * We removed undefined values in this function as `json-logic` ignores them.
- * Means we will always check against a value for validations.
+ * jsonLogic interprets  undefined and null values differently when running comparisons and that creates inconsistent results.
+ * This function attempts to fix that (ported from v0).
  *
  * @param {object} values - a set of values from a form
  * @returns {object} values object without any undefined
  */
-function replaceUndefinedValuesWithNulls(values: any = {}) {
+function replaceUndefinedAndNullValuesWithNaN(values: any = {}) {
   return Object.entries(values).reduce((prev, [key, value]) => {
     return { ...prev, [key]: value === undefined || value === null ? Number.NaN : value }
```

Comment:
suggestion: I think this function would be readable without the reducer and instead we use direct object assignment:

```ts
function replaceUndefinedAndNullValuesWithNaN(values: any = {}) {
  const result = {};
  for (const [key, value] of Object.entries(values)) {
    result[key] = value === undefined || value === null ? Number.NaN : value;
  }
  return result;
}
```

Reply:
Sounds good to me 👍